### PR TITLE
ADEN-3800 Fix mobile PageFair

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/pageFairDetection.js [hunk 1]
```diff
@@ -50,8 +50,19 @@ define('ext.wikia.adEngine.pageFairDetection', [
 		dispatchDetectionEvent(eventName);
 	}
 
+	function canBeInitialized(injectedContext) {
+		return injectedContext !== undefined &&
+			   injectedContext.hasOwnProperty('opts') &&
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/pageFairDetection.js]: why not  `injectedContext && injectedContext.opts && injectedContext.opts.pageFairDetection`?

File: extensions/wikia/AdEngine/js/pageFairDetection.js [hunk 2]
```diff
@@ -50,8 +50,19 @@ define('ext.wikia.adEngine.pageFairDetection', [
...
+			   injectedContext.opts.hasOwnProperty('pageFairDetection') &&
+			   injectedContext.opts.pageFairDetection;
+	}
+
 	function initDetection(injectedContext) {
 		var node = doc.getElementsByTagName('script')[0];
+
+		if (!canBeInitialized(injectedContext)) {
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/pageFairDetection.js]: minor: what this method does is arguments validation, consider renaiming

File: extensions/wikia/AdEngine/js/run/desktop.run.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ require([
 
 		sourcePoint.initDetection();
 
-		if (context.opts.pageFairDetection) {
+		if (pageFair.canBeInitialized(context)) {
 			pageFair.initDetection(context);
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/run/desktop.run.js:70]: you are doing the check twice, one check is here and the second one (for the same variable) is inside initDetection. Consider simplification.

File: extensions/wikia/AdEngine/js/spec/pageFairDetection.spec.js [hunk 4]
```diff
@@ -80,18 +80,22 @@ describe('Module ext.wikia.adEngine.pageFairDetection', function () {
 		expect(mocks.window.ads.runtime.pf.blocking).toBe(false);
 	});
 
+	it('should throw an exception if its try to initialize without correct context', function () {
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/spec/pageFairDetection.spec.js]: `should throw an exception if it tries to initialize without correct context`

File: extensions/wikia/AdEngine/js/pageFairDetection.js [hunk 5]
```diff
@@ -60,6 +71,7 @@ define('ext.wikia.adEngine.pageFairDetection', [
 	}
 
 	return {
+		canBeInitialized: canBeInitialized,
```
Comment (drets) [inline extensions/wikia/AdEngine/js/pageFairDetection.js]: I guess you exposed this method basically for fixing issue on Mercury...
1. I am curious why not use `adContext` directly on Mercury?
2. Do we have the same issue (enabled detection by default) on Mercury for _SourcePoint_?

```
            if (Ads.previousDetectionResults.sourcePoint.exists) {
                this.trackBlocking('sourcePoint', this.GASettings.sourcePoint, Ads.previousDetectionResults.sourcePoint.value);
            } else {
                this.sourcePointDetectionModule.initDetection();
            }

            if (Ads.previousDetectionResults.pageFair.exists) {
                this.trackBlocking('pageFair', this.GASettings.pageFair, Ads.previousDetectionResults.pageFair.value);
 -          } else {
 +          } else if (this.pageFairDetectionModule.canBeInitialized(adsContext)) {
                this.pageFairDetectionModule.initDetection(adsContext);
            }
```

File: extensions/wikia/AdEngine/js/run/desktop.run.js (hunk 3 above)
Comment (bognix) [inline extensions/wikia/AdEngine/js/run/desktop.run.js:70]: and you are doing the check one more time in Mercury. Isn't the check inside initDetection sufficient?

File: extensions/wikia/AdEngine/js/pageFairDetection.js (hunk 5 above)
Comment (v3rth) [inline extensions/wikia/AdEngine/js/pageFairDetection.js]: I wanted to validate context inside initDetection method and additionally allow to check it outside
1. probably I will do it in next change 
2. nope, SourcePoint has check-method inside initDetection

Comment (drets) [inline extensions/wikia/AdEngine/js/pageFairDetection.js]: Consider to do it in the same way as _SourcePoint_ (for keeping consistency).

**Pull request closed without merging**